### ISSUE TITLE
Question about Live Looping functionality status & plans

### ISSUE THREAD

**Issue by Gimmeapill** [Buzztrax/buzztrax#49]
Looking at the roadmap for 0.9 and the TODO, there's mention of a Buzztrax-Live module
http://wiki.buzztrax.org/index.php/Buzztrax-live
and remote control:
http://wiki.buzztrax.org/index.php/Remote_control

Just wondering how  far you are in the design, as I'm looking eagerly for live looping.
Is the Playback controller intended to provide equivalent functionality as the BTDSys's "LiveJumpHACK"?
http://cdn7.atwikiimg.com/_pub/buzzusers/backup/BuzzMachines_help_backup/BTDSys%20LiveJumpHACK.html

BR,

LX

**Comment by ensonic** [comment]
This is cool for sure. Right now, the playback controller supports the classic play/pause and seeking. But using keys to jump to direct positions is a nice idea. I'll think about how to do that, but we'll definitely want to have this.

**Comment by Gimmeapill** [comment]
Yeah, BTD surely had a genius idea at the time, LiveJumpHACK was pretty rough, but for a hack machine it worked flawlessly (it served me well for 10+ years) - used in combination with PeerState (for automating preset changes) and Peerctrl (for advanved midi mapping) it gave old Buzz instant live abilities. 
In a modern application, I'm not sure what would make more sense though. You may find some ideas in the other Buzz clones around: Paniq did make a try in Aldrin but that never worked properly. Oskari, on his side, implemented a machine conceptually closer to ableton in NuBuzz called Jeskola Live - but I guess this one would fit more with Buzztrax-Live than the remote control, as it looks like a complete new interface. 
Buze seems also have similar custom looping capabilities, altough I never really managed to understand its sequencer.

Regarding "basic" live looping as LiveJump does it, here's what I understand would be needed for the playback controller:
- a way to define loop section boundaries (start and stop point in ticks?)
- then a way to bind each loop section to a key/midi cc/note
- a setting to define how long to wait before jumping when a trigger is received: instantly/ at the end of the currently playing loop/ after an arbitrary number of ticks. If no trigger is received, it should probably loop indefinitely in the currently playing section.
- some kind of visual feedback when running. LiveJump has a nice counter window that decrement until the next "Jump" - NASA style ;-)
But maybe that also could be the kind of thing to run without any GUI in headless mode - sending back feedback via OSC or midi to the controller

Anyway, just adding my 2 cent as I saw the blank wiki pages - Buzztrax is getting nicely is shape btw ;-)

Cheers,

LX

**Comment by ensonic** [comment]
I think I'll have to play with LiveJumpHACK. Too bad there is no video on youtube showcasing it :/

**Comment by Gimmeapill** [comment]
Well, I'm quite retired from the live electronic music scene nowadays - so my use case has all but faded away. But if you have some time to dedicate to the matter, I'll try to dust off my Buzz live set and take a smartphone video of LiveJumpHACK in action. Renoise also does something relatively similar when pattern looping mode is enabled.

**Comment by ensonic** [comment]
That would be awesome!

**Comment by Gimmeapill** [comment]
Hi Stefan,
Sorry, but I still didn't have a chance to follow up on this one. My Buzz/wine/wineasio installation is busted, and I've been too lazy to set it up again...so no video for now. 
But I don't think you miss much: there's nothing to see when it runs except a countdown window grabbing keyboard input (midi doesn't need that window), so the documentation page will probably be more helpful if you want so implement something similar:
http://img.atwikiimg.com/www7.atwiki.jp/buzzusers/pub/backup/BuzzMachines_help_backup/BTDSys%20LiveJumpHACK.html
I also noticed that BTDSys still has an online presence: http://btdsys.lazytrap.com/
You could maybe drop him a line to get the source?

**Comment by ensonic** [comment]
You are right, I'll get it from the html page. What I could do in buzztrax, is to add shortcuts to sequence labels (as a first step).
